src/components/list/list.scss

```diff
@@ -295,7 +295,15 @@ md-list-item {
       align-items: center;
 
       height: 100%;
-      margin: auto;
+
+      // The secondary container is now static and needs to be aligned at the end of its parent.
+      // - Using an absolute position will cause much issues with the overflow.
+      // - Using a flex-filler, which pushes the container to the end of the parent is working
+      //   but breaks the users list-item layout.
+      // Using margin auto to move them to the end of the list item is more elegant, because it has
+      // a lower priority than the flex filler and isn't introducing overflow issues again.
+      // The margin on the top is also important to align multiple secondary items vertical correctly.
+      @include rtl(margin, auto 0 auto auto, auto auto auto 0);
```

Comment:
Both approaches are working, but I will change it to yours, because it seems IMO more clear.